lc-hax/Scripts/Modules/TriggerMod.cs

```diff
@@ -92,6 +92,49 @@ public sealed class TriggerMod : MonoBehaviour, IEnemyPrompter {
                 doorLock.UnlockDoorSyncWithServer();
                 break;
             }
+            if (collider.TryGetComponent(out NoisemakerProp NoiseMaker)) {
+                if (!NoiseMaker.IsOwner) {
+                    if (Helper.LocalPlayer == null) break;
+                    NoiseMaker.ChangeOwnershipOfProp(Helper.LocalPlayer.actualClientId);
+                }
+
+                NoiseMaker.ItemActivate(true, true);
+                break;
+            }
+            if (collider.TryGetComponent(out BoomboxItem boombox)) {
+                if (!boombox.IsOwner) {
+                    if (Helper.LocalPlayer == null) break;
+                    boombox.ChangeOwnershipOfProp(Helper.LocalPlayer.actualClientId);
+                }
+                boombox.ItemActivate(true, true);
+                break;
+            }
+            if (collider.TryGetComponent(out WhoopieCushionItem fartcushion)) {
+                fartcushion.Fart();
+                break;
+            }
+            if (collider.TryGetComponent(out AnimatedItem animated)) {
+                if (!animated.IsOwner) {
+                    if (Helper.LocalPlayer == null) break;
+                    animated.ChangeOwnershipOfProp(Helper.LocalPlayer.actualClientId);
+                }
+                animated.EquipItem();
+                break;
+            }
+
+            if (collider.TryGetComponent(out GrabbableObject prop)) {
+                if (!prop.IsOwner) {
+                    if (Helper.LocalPlayer == null) break;
+                    prop.ChangeOwnershipOfProp(Helper.LocalPlayer.actualClientId);
+                }
+                if (prop.GetComponent<BoomboxItem>() != null
+                    || prop.GetComponent<NoisemakerProp>() != null
+                    || prop.GetComponent<AnimatedItem>() != null
+                    || prop.GetComponent<WhoopieCushionItem>() != null) break;
```

Comment:
Make it a habit to use `TryGetComponent` as much as possible. We are working with compiled Unity which doesn't work the same way in the editor. Sometimes references will go missing and continue to go missing even with null checks. `TryGetComponent` is a more robust way to prevent these problems later.

Reply:
If it is, I wouldn’t have to create ResilientCoroutine and I would’ve continue using `GetComponent` since it’s much simpler.

Reply:
Fixed and moved in a grabbable extension.

Reply:
Looks a lot cleaner.